stan/math/prim/scal/err/check_2F1_converges.hpp

```diff
@@ -58,9 +58,12 @@ namespace stan {
 
       bool is_undefined = is_nonpositive_integer(b1) && fabs(b1) <= num_terms;
 
-      if (is_polynomial && !is_undefined) return;
-      if (fabs(z) < 1.0 && !is_undefined) return;
-      if (fabs(z) == 1.0 && !is_undefined && b1 > a1 + a2) return;
+      if (is_polynomial && !is_undefined)
```

Comment:
Decide this one shouldn't be a big and/or of 

```
(!is_undefined
 && (is_polynomial
     || fabs(z) < 1
     || fabs(z) == 1 && b1 > a1 + a2))
```
The way you have it seems ab it clearer, but my suggestion is more efficient because unless `!is_undefined`, it will fail early.

. The way you have it here is probably clearer, but it's a bit inefficient to do all these tests if `is_undefined` is false, whereas early failure will avoid all the tests in the exapnded form